src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -606,6 +607,11 @@ private User impersonate(final RestRequest request, final User originalUser) thr
             // loop over all http/rest auth domains
             for (final AuthDomain authDomain : restAuthDomains) {
                 final AuthenticationBackend authenticationBackend = authDomain.getBackend();
+
+                if (authDomain.getHttpAuthenticator() instanceof OnBehalfOfAuthenticator) {
+                    continue;
```

Comment:
Cuz we would like to skip the OBO authbackend when we trying to impersonate the user. Otherwise it will cause the trouble, since the OBO authbackend could not handle some of the user information directly from the User class.

Reply:
We should have this documented as a comment in the code, otherwise we won't know why this line is here after this is merged.

Reply:
added